pkg/validation/internal/bundle.go

```diff
@@ -24,71 +22,82 @@ func validateBundles(objs ...interface{}) (results []errors.ManifestResult) {
-	spec := operatorsv1alpha1.ClusterServiceVersionSpec{}
-	if err := json.Unmarshal(bcsv.Spec, &spec); err != nil {
-		return nil, errors.ErrInvalidParse(fmt.Sprintf("converting bundle CSV %q", bcsv.GetName()), err)
+func validateOwnedCRDs(bundle *registry.Bundle, csv *registry.ClusterServiceVersion) (result errors.ManifestResult) {
+	ownedKeys, _, err := csv.GetCustomResourceDefintions()
+	if err != nil {
+		result.Add(errors.ErrInvalidParse("error getting CSV CRDs", err))
+		return result
 	}
-	return &operatorsv1alpha1.ClusterServiceVersion{
-		TypeMeta:   bcsv.TypeMeta,
-		ObjectMeta: bcsv.ObjectMeta,
-		Spec:       spec,
-	}, errors.Error{}
-}
 
-func validateOwnedCRDs(bundle *registry.Bundle, csv *operatorsv1alpha1.ClusterServiceVersion) (result errors.ManifestResult) {
-	ownedCrdNames := getOwnedCustomResourceDefintionNames(csv)
-	crdNames, err := getBundleCRDNames(bundle)
-	if err != (errors.Error{}) {
-		result.Add(err)
+	keySet, rerr := getBundleCRDKeys(bundle)
+	if rerr != (errors.Error{}) {
+		result.Add(rerr)
 		return result
 	}
 
-	// validating names
-	for _, crdName := range ownedCrdNames {
-		if _, ok := crdNames[crdName]; !ok {
-			result.Add(errors.ErrInvalidBundle(fmt.Sprintf("owned CRD %q not found in bundle %q", crdName, bundle.Name), crdName))
+	// Validate all owned keys, and remove them from the set if seen.
+	for _, ownedKey := range ownedKeys {
+		if _, ok := keySet[*ownedKey]; !ok {
+			result.Add(errors.ErrInvalidBundle(fmt.Sprintf("owned CRD %s not found in bundle %q", keyToString(*ownedKey), bundle.Name), *ownedKey))
 		} else {
-			delete(crdNames, crdName)
+			delete(keySet, *ownedKey)
 		}
 	}
 	// CRDs not defined in the CSV present in the bundle
-	for crdName := range crdNames {
-		result.Add(errors.WarnInvalidBundle(fmt.Sprintf("owned CRD %q is present in bundle %q but not defined in CSV", crdName, bundle.Name), crdName))
+	for key := range keySet {
+		result.Add(errors.WarnInvalidBundle(fmt.Sprintf("CRD %s is present in bundle %q but not defined in CSV", keyToString(key), bundle.Name), key))
 	}
 	return result
 }
 
-func getOwnedCustomResourceDefintionNames(csv *operatorsv1alpha1.ClusterServiceVersion) (names []string) {
-	for _, ownedCrd := range csv.Spec.CustomResourceDefinitions.Owned {
-		names = append(names, ownedCrd.Name)
-	}
-	return names
-}
-
-func getBundleCRDNames(bundle *registry.Bundle) (map[string]struct{}, errors.Error) {
+func getBundleCRDKeys(bundle *registry.Bundle) (map[registry.DefinitionKey]struct{}, errors.Error) {
 	crds, err := bundle.CustomResourceDefinitions()
 	if err != nil {
 		return nil, errors.ErrInvalidParse("error getting bundle CRDs", err)
 	}
-	crdNames := map[string]struct{}{}
+	keySet := map[registry.DefinitionKey]struct{}{}
 	for _, crd := range crds {
-		crdNames[crd.GetName()] = struct{}{}
+		if len(crd.Spec.Versions) == 0 {
+			// Skip group, which CSVs do not support.
+			key := registry.DefinitionKey{
+				Name:    crd.GetName(),
```

Comment:
Our work adding v1 crds to bundles is going to break this - instead of returning a `*v1beta1.CustomResourceDefinition` we are returning an interface now. We started addressing this in https://github.com/operator-framework/api/pull/29/files#diff-8a818e661938b6d25ef2013d1289f89cR94
but I believe we are planning to move this code back into the registry instead - right @dinhxuanvu? 

See https://github.com/operator-framework/operator-registry/pull/295 for the PR with the breaking change. we are planning to merge this one this week. 

We can merge as-is and then fix it in a follow-up, or hold off on this change.

Reply:
This PR is meant to fix api as it is. We will address the v1 CRD support in subsequent PR.

Reply:
Ok, sounds good